include/fc/io/cfile.hpp

```diff
@@ -0,0 +1,141 @@
+#pragma once
+#include <fc/filesystem.hpp>
+#include <cstdio>
+
+namespace fc {
+
+namespace detail {
+   using unique_file = std::unique_ptr<FILE, decltype( &fclose )>;
+}
+
+class cfile_datastream;
+
+/**
+ * Wrapper for c-file access that provides a similar interface as fstream without all the overhead of std streams.
+ * std::runtime_error exception thrown for errors.
+ */
+class cfile {
+public:
+   cfile()
+     : _file(nullptr, &fclose)
+   {}
+
+   void set_file_path( fc::path file_path ) {
+      _file_path = std::move( file_path );
+   }
+
+   fc::path get_file_path() const {
+      return _file_path;
+   }
+
+   bool is_open() const { return _open; }
+
+   size_t file_size() const {
+      return fc::file_size( _file_path );
+   }
+
+   /// @param mode is any mode supported by fopen
+   ///        Tested with:
+   ///         "ab+" - open for binary update - create if does not exist
+   ///         "rb+" - open for binary update - file must exist
+   void open( const char* mode ) {
+      _file.reset( fopen( _file_path.generic_string().c_str(), mode ) );
```

Comment:
fopen() has some weirdness in win32 if wanting to support unicode filenames. See EOSIO/eos#7640 for what I did in the only other place we've used fopen. Not sure if better solution exists

Reply:
Added similar fix as #7640